### PR TITLE
improve readability

### PR DESCRIPTION
**Pull request by maximlopin**
- make code in `is_foreign_key()` easier to read

### CODE REVIEW

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

File: mindsdb_native/libs/helpers/text_helpers.py [hunk 1]
```diff
@@ -142,37 +142,39 @@ def splitRecursive(word, tokens):
...
-            if not (is_uuid and is_same_length):
-                data_looks_like_id = False
-                break
+    # Detect UUID
+    if foregin_key_type:
+        is_uuid = False
+    else:
+        all_same_length = all(len(str(data[0])) == len(str(x)) for x in data)
+        uuid_charset = set('0123456789abcdef-')
+        all_uuid_charset = all(set(str(x)).issubset(uuid_charset) for x in data)
```
Comment (paxcema) [inline mindsdb_native/libs/helpers/text_helpers.py:175]: If I'm reading this right, `str(x).lower()` would also capture UUID's in all caps.

**Review by paxcema — APPROVED** [review]

File: mindsdb_native/libs/helpers/text_helpers.py (hunk 1 above)
Comment (maximlopin) [reply mindsdb_native/libs/helpers/text_helpers.py:175]: changing `set('0123456789abcdef-')` to `set('0123456789abcdefABCDEF-')` would solve that, right?

Comment (paxcema) [reply mindsdb_native/libs/helpers/text_helpers.py:175]: Yup, that works too!

**Comment by mindsdbadmin** [comment]
**CLA Assistant Lite** All Contributors have signed the CLA.

**Pull request merged**